src/gui/app.cpp

```diff
@@ -12,13 +13,14 @@ App::App(int &argc, char ** argv)
 bool App::event(QEvent *event)
 {
   if( event->type()  == QEvent::FileOpen) {
+    //QFileOpenEvent is triggered for each file|URL seperately. Kepping
+    //the QList<QUrl> here does not cause any file|URL to be ignored.
+    QList<QUrl> urls;
     QFileOpenEvent * fileOpenEvent = static_cast<QFileOpenEvent *>(event);
     if(fileOpenEvent) {
       QString file = fileOpenEvent->file();
-      if(!file.isEmpty()) {
-        emit neovimOpenFileTriggered(file);
-        return true;
-      }
+      urls.append(fileOpenEvent->url());
+      emit openFilesTriggered(urls);
```

Comment:
You should be able to replace the urls variable above and these two lines with
```C++
emit openFilesTriggered(QList<QUrl>() << fileOpenEvent->url());
```

or (c++11)

```C++
emit openFilesTriggered({fileOpenEvent->url()});
```